cloud_telemetry/middleware.py

```diff
@@ -0,0 +1,71 @@
+"""Middleware for OpenTelemetry instrumentation."""
+
+from opentelemetry import metrics, trace
+from opentelemetry.exporter.otlp.proto.http.metric_exporter import OTLPMetricExporter
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.instrumentation.starlette import StarletteInstrumentor
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from starlette.applications import Starlette
+
+
+class OpenTelemetryInstrumentor:
+    """Instruments a Starlette application with OpenTelemetry."""
+
+    @staticmethod
```

Comment:
Any reason for `staticmethod` specifically? 

I think this ties us down from making custom choices, I think we would might want these methods to interact with other data and memebers, for example lets say you want to tracing and metric boolean to be defined at class level and then set the dafault from class member, this way user can have more flexibility when envoking these methods, unless ofc you are aiming for some desing pattern like `factory patter` or `singleton patter`.

Reply:
Nice, I have pushed some changes, please have a look.